components/CreateWorkout/CreateWorkout.jsx

```diff
@@ -37,9 +38,60 @@ const AddCycleButton = styled(PlusButton)`
    right: 25px;
 `;
 
-export default ({ navigation }) => {
-  const [name, setName] = React.useState('');
-  const [itemState, setItemState] = React.useState([]);
+const getExercise = (exercises, exerciseId) => _.find(exercises, (exercise) => exercise.id === exerciseId);
+
+const parseItemStateToWorkout = (itemState) => {
+  // Get two unique muscleGroups
+  const allMuscleGroups = itemState.map((item) => item.muscleGroups);
+  const workoutMuscleGroups = _.uniq(allMuscleGroups);
+  const topMuscleGroups = workoutMuscleGroups.slice(0, 2).join(' ');
+
+  return {
+    muscleGroups: topMuscleGroups,
+    exercises: itemState.map((item) => {
+      const setArr = [];
+      const sets = item.sets || 3;
+      for (let i = 0; i < sets; i += 1) {
+        if (item.isReps) {
+          setArr.push({ weight: null, reps: item.reps || 10 });
+        } else {
+          setArr.push({ weight: null, time: item.seconds || 60 });
+        }
+      }
+      return {
+        sets: setArr,
+        exerciseId: item.id,
+      };
+    }),
+  };
+};
+
+export default ({ navigation, route }) => {
+  const isEditing = _.get(route, ['params', 'editing'], false);
+  const editWorkout = _.get(route, ['params', 'editWorkout'], undefined);
+
+  const exercises = useSelector((state) => state.exercises.exercises);
+
+  const parseWorkoutExercises = (exercise, index) => {
+    const reps = _.get(exercise, ['sets', '0', 'reps'], '').toString();
+    const seconds = _.get(exercise, ['sets', '0', 'time'], '').toString();
+
+    const matchingExercise = getExercise(exercises, exercise.exerciseId);
+
+    return {
+      color: COLORS[index % COLORS.length],
+      id: exercise.exerciseId,
+      name: matchingExercise.name,
+      muscleGroups: matchingExercise.muscleGroups,
+      sets: exercise.sets.length.toString(),
+      reps,
+      seconds,
+      isReps: !!reps,
+    };
+  };
+
+  const [name, setName] = React.useState(_.get(editWorkout, ['name'], ''));
+  const [itemState, setItemState] = React.useState(editWorkout ? editWorkout.exercises.map(parseWorkoutExercises) : []);
```

Comment:
should the map be done outside of the initial state? Won't it try and map every time this component re renders

Reply:
Nope, actually the initial state code only runs the very first time the component is created, so this code will only map once